### ISSUE TITLE
MMS failed send

### ISSUE THREAD

**Issue by brittupton** [moezbhatti/qksms#1743]
### DESCRIPTION
no matter what i do, no images will send in my QKSMS app
i can send stickers and gifs from my keyboard just fine but if i try to send an image from my gallery app it won't even try to send it. the message below the image immediately goes to "failed to send. tap to try again."
i tap to try again and it tries for a split second then reverts back to the same error

### STEPS
1. open QKSMS
2. choose recipient
3. attach image from gallery
4. tap send
5. error message

### EXPECTED
images from the gallery will be sent to the recipient

### OBSERVATIONS
only receive the error code below the image

**Comment by r0xsh** [comment]
I had this problem too. Some phone operators have a limit on the size of attachments. In the qksms settings there is an option to automatically compress the sent files. By changing this option my mms have been re-functioning.

**Comment by brittupton** [comment]
@r0xsh i tried changing them to almost every one and "automatic" seemed to work maybe CD or twice but didn't stick :/

**Comment by Trailhead-Jared** [comment]
As one more data point. I noticed this problem since updating to 3.9.2 Setting it to automatic seemed to fix it. 2000KB and no-compression (the old value) didn't seem to work

**Comment by unresolvedsymbol** [comment]
Bump? On 3.9.4 and can't use the presrt 1000kb compression for some reason even though that's my carriers official limit. I think the best solution to this would to be allow a custom setting? Ex.  I could try 950kb and see if there was MMS metadata in the way or something that would push it over the limit.

**Comment by e-t-l** [comment]
I was also having trouble consistently getting MMS to send (v3.9.4), and I tried a bunch of different combinations of settings and here's what I found:
1. It seemed like there may be an issue with the Automatic compression; when I set compression to a specific value (anything from 100KB to 1000KB), it was more likely to send successfully.
2. I think the biggest issue was my adblocker. I'm using [Nebulo](https://git.frostnerd.com/PublicAndroidApps/smokescreen), a DNS-based adblock VPN app similar to [Blokada](https://github.com/blokadaorg/blokada). I noticed the key-symbol that indicates VPN connection was flashing on and off in my statusbar when I tried to send an MMS, so on a hunch I disabled Nebulo, and suddenly my MMS were sending much more reliably. So, I'm guessing Nebulo was interfering with some aspect of Android's MMS-sending process. Unfortunately, I have no idea what process, service, or domain it might be blocking. Nebulo _does_ have an option to whitelist, including system apps and services. Does anyone know what Android System apps/services might be involved in how QKSMS processes MMS?

EDIT: I previously thought switching to a different messaging app solved the issue too, but on further investigation I'm still getting issues sending MMS with Nebulo active, so Nebulo is definitely the issue.I'm going to go open an issue with them, but it would still be helpful if anyone knows the names of the android system services that might be involved with sending MMS.

**Comment by e-t-l** [comment]
UPDATED FINDINGS

I think the compression and the vpn adblocker are two completely separate issues now. My phone is still seeing MMS very inconsistently, but these are the findings I can reproduce _most_ consistently:

1. When starting QKSMS for the first time (e.g. after restarting phone or force-quitting and clearing cache of QKSMS), if compression is set to Automatical, MMS fail to send immediately. If I change compression level (e.g. to 1000KB) and send an MMS, it succeeds. If I then switch compression _back_ to Automatic, MMS will continue to send successfully. 
2. If Nebulo is active, MMS will fail to send regardless of messaging app (well, I only tried QKSMS and Signal-as-unencrypted-SMS/MMS, since those are the ones I have installed). This occurs even if Nebulo does not log that any domain requests were blocked. Unlike the other case I described where the MMS says "failed to send" immediately, with Nebulo active it says "Sending..." for ~30 seconds before failing. 

In conclusion, I have no idea what's going on. Despite no updates, to my knowledge, to either QKSMS or Nebulo (or Android's MMS handler services), this is a fairly recent issue for me.

**Comment by unresolvedsymbol** [comment]
I use adaway (/etc/hosts based blocker) but I'm definitely sure my carriers MMS domains aren't blocked since it works at 600kb. It must be metadata, I haven't tried specific values as I don't use my SIM in an Android anymore and the current static values 600kb and 1000kb would require me to recompile the app.

**Comment by e-t-l** [comment]
Good addition. Just to test it, I replaced Nebulo with Blokada and it doesn't cause MMS to fail, so I may just keep Blokada.Shame, Nebulo's a nice lightweight app, but it's only maintained by 1 person and it's just too buggy.

Still no fix for the compression issue, but it's interesting that switching away from Automatic compression then switching back seems to be a decent workaround.

**Comment by xgorex** [comment]
I have the same / similar issue. My carrier does not limit the MMS size. I can easly receive 2.5MB MMS sent by my wife Moto One. Unfortunatelly I cant sent MMS bigger than 1000kB with QKSMS. Now, if I switch to LineageOS stock SMS app, it can send 2000kB MMS on the same phone/SIM.

**Comment by crahn** [comment]
Could this be fixed in here?

MMS not sending (c0b2f61)

Are you at 3.8.1 or higher?

Release notes here show similar issues being fixed.

https://github.com/moezbhatti/qksms/releases/tag/v3.8.1

**Comment by e-t-l** [comment]
@crahn I'm on v3.9.4 and still experiencing the issues with MMS compression.
